### DISCUSSION TITLE
NetworkForAGraphWithAttributes Mini-Batching Bug

### DISCUSSION THREAD

**Discussion by ColinKohler** [e3nn/e3nn#388] [General]
The current implementation of v2106.gate_points_networks.NetworkForAGraphWithAttributes is a bit at odds with pytroch-geometric's best practices. 

In pytorch-geometric's Data class edges are specified using the 'edge_index' field whereas NetworkForAGraphWithAttributes expects 'edge_src' and 'edge_dst' fields. This causes issues with mini-batching as in pytroch-geometric, the values in edge_index are automatically incremented by the cumulated number of nodes of all graph when creating a mini-batch [(see here)](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html). At the moment NetworkForAGraphWithAttributes has no such incrementing and therefor you'd have to create a custom DataLoader or Data class to handle this.

If there is no reason that I'm missing as to why this was written this way I would like to update NetworkForAGraphWithAttributes to use 'edge_index' for predefined graphs. I have a branch which does this and can submit a PR if desired.

Thanks!

**Comment by mariogeiger** [comment]
Sure feel free to propose a PR, I m not sure I fully understood the modification suggestions, just make a PR and I'll have a look with pleasure

**Reply by ColinKohler** [reply]
[PR](https://github.com/e3nn/e3nn/pull/389)
